### ISSUE TITLE
How to load/save presets with Facets Dive

### ISSUE THREAD

**Issue by plopd** [PAIR-code/facets#173]
<img width="1176" alt="screenshot 2019-01-30 at 11 00 12" src="https://user-images.githubusercontent.com/1534513/51973778-b739ab80-247e-11e9-983e-84593ba338d7.png">

How can I provide the presets prior to the visualisation (What DOM elements do I need to access and what are the data types?)

How do retrieve the current presets and save it?

As a general note, Facets Dive is awesome but it lacks a clear API that makes it easy for non-JS (but Python users) to use it.

**Comment by jameswex** [comment]
Currently there is no direct way to in python set/get presets on the Facets Dive vis. In JS, the data attributes of the facets-dive element that you can set/get for this are found here: https://github.com/PAIR-code/facets/tree/master/facets_dive#interactive-properties

You could set these properties in the same HTML/JS code that you use to display dive in an output cell like adding to the notebook code here https://colab.sandbox.google.com/github/PAIR-code/facets/blob/master/colab_facets.ipynb#scrollTo=XtOzRy8Z3M36 something like  `document.querySelector("#elem").horizontalFacet = 'Age';`

It should be possible for us (or anyone else) to make a python wrapper that creates and publishes the HTML and has python methods to set and get those properties.

**Comment by plopd** [comment]
Programmatically setting presets is possible, see: https://rare-technologies.com/sharing-facets-dive-visualizations/

How is getting presets?
I see at least two options:

1. 

Either it is possible to do it with current Dive or

2. I hide the dropdown toolbar menu from Dive and implement my own toolbar. The question here is then: How do I hide the dropdown toolbar menu?

**Comment by jameswex** [comment]
In javascript you can get the presets with "var hFacets = document.querySelector("#elem").horizontalFacet;" Then you would need to pass that preset back to python if that is where you are looking to use it. Here is one way to provide javascript vars back to python in jupyter: https://stackoverflow.com/questions/37172978/how-to-pass-variables-from-javascript-to-python-in-jupyter

**Comment by plopd** [comment]
It turns out that for getting the user's presets it is easier if I would hide the Dive's dropdown menu (Binning on X-axis, ...) and create my own. My question is how do I hide the dropdown menu?

**Comment by jameswex** [comment]
You can hide that control with a change to HTML_TEMPLATE in the example notebook like:
`
HTML_TEMPLATE` = """<link rel="import" href="https://raw.githubusercontent.com/PAIR-code/facets/master/facets-dist/facets-jupyter.html">
        <facets-dive id="elem" height="600"></facets-dive>
        <script>
          var data = {jsonstr};
          document.querySelector("#elem").data = data;
          setTimeout(function() {{
            document.querySelector("facets-dive-controls").style.display = 'none';
          }}, 1000);
        </script>"""
`
Basically, you can set the actual controls element to have display setting of "none", but you need to wait until the entire polymer element has been rendered first, hence the setTimeout. A better solution is to check in setTimeout that document.querySelector("facets-dive-controls") is not null. If it is null, then call setTimeout again to wait some amount of time before trying again.
